kuksa-sdk/src/test/kotlin/org/eclipse/kuksa/connectivity/databroker/docker/DockerDatabrokerContainer.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2023 - 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa.connectivity.databroker.docker
+
+import com.github.dockerjava.api.DockerClient
+import com.github.dockerjava.api.command.CreateContainerResponse
+import com.github.dockerjava.api.command.PullImageResultCallback
+import com.github.dockerjava.api.command.WaitContainerResultCallback
+import com.github.dockerjava.api.exception.NotModifiedException
+import com.github.dockerjava.api.model.ExposedPort
+import com.github.dockerjava.api.model.HostConfig
+import com.github.dockerjava.api.model.InternetProtocol
+import com.github.dockerjava.api.model.PortBinding
+import com.github.dockerjava.api.model.Ports
+import com.github.dockerjava.core.DefaultDockerClientConfig
+import com.github.dockerjava.core.DockerClientImpl
+import com.github.dockerjava.httpclient5.ApacheDockerHttpClient
+import org.eclipse.kuksa.connectivity.databroker.DATABROKER_TIMEOUT_SECONDS
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+private const val KEY_PROPERTY_DATABROKER_TAG = "databroker.tag"
+private const val KEY_ENV_DATABROKER_TAG = "DATABROKER_TAG"
+private const val DEFAULT_DATABROKER_TAG = "master"
+
+/**
+ * Starts and stops the Databroker Docker Container. Per default the image with the master tag is pulled and started.
+ * The version of the image can be influenced by either a System Property with value "databroker.tag" or an
+ * Environment Variable with value "DATABROKER_TAG".
+ *```
+ * Samples
+ *
+ * System Property:
+ * ./gradlew clean build -Ddatabroker.tag="0.4.1"
+ *
+ *
+ * Environment Variable:
+ * export DATABROKER_TAG="0.4.1"
+ * ./gradlew clean build
+ *
+ * or
+ *
+ * DATABROKER_TAG="0.4.1" ./gradlew clean build
+ * ```
+ */
+abstract class DockerDatabrokerContainer(
```

Comment:
Shouldn't the class be called DataBrokerDockerContainer? Because if we would have an interface class like "DockerContainer" and then implementing it for a DataBroker -> "DataBrokerDockerContainer"